ste/mgr-JobPartTransferMgr.go

```diff
@@ -186,6 +185,21 @@ func (jptm *jobPartTransferMgr) GetSourceCompressionType() (common.CompressionTy
 	return common.GetCompressionType(encoding)
 }
 
+func (jptm *jobPartTransferMgr) getSrcUserDelegationSAS(src string) (string, error) {
+	// panics if udam isn't found
+	udam := jptm.jobPartMgr.GetUserDelegationAuthenticationManagerInstance()
+
+	objectURL, err := url.Parse(src)
+
+	if err != nil {
```

Comment:
TODO: add note that if this fails to parse, something has gone terribly, terribly wrong.